### ISSUE TITLE
Evaluation of pretrained model with Gutenberg dataset on NSP task

### ISSUE THREAD

**Issue by batrlaluk** [google-research/bert#380]
We are trying to evaluate the pretrained model on data from Gutenberg (around 3000 books) but we are not able to get anyway close to the result from the paper where you achieved 97-98 % for the Next Sentence Prediction (NSP) task.

We did some basic data preprocessing to get each sentence on a single line with empty lines between paragraphs. Then we run the `create_pretraining_data.py` and `run_pretraining.py` where we set `--do_train=False` (see the scripts below). This way we achieved around 78 % accuracy for the next sentence prediction task, which is 20 % less than what you achieved in the paper.

Is our approach correct or are we doing something wrong?
Could you share the algorithm you used to preprocess the books from the BookCorpus dataset? Maybe the way we preprocess the data is what causes the difference.

`python create_pretraining_data.py   --input_file=../Gutenberg/sentence_per_line/*.txt   --output_file=../Gutenberg/output/tokenized_tf_examples.tfrecord   --vocab_file=$BERT-BASE/vocab.txt   --do_lower_case=True   --max_seq_length=128   --max_predictions_per_seq=20   --masked_lm_prob=0.15   --random_seed=12345   --dupe_factor=5`

`python run_pretraining.py   --input_file=../Gutenberg/output/tokenized_tf_examples.tfrecord   --output_dir=../Gutenberg/output/pretraining_output   --do_train=False   --do_eval=True   --bert_config_file=$BERT-BASE/bert_config.json   --init_checkpoint=$BERT-BASE/bert_model.ckpt   --train_batch_size=32   --max_seq_length=128   --max_predictions_per_seq=20   --num_train_steps=20   --num_warmup_steps=10   --learning_rate=2e-5`

**Comment by maxlund** [comment]
`--num_train_steps` should be higher, i think it says 10000 or more in the readme.
